### ISSUE TITLE
`def fn(x: Type[T]) -> T:` not handled

### ISSUE THREAD

**Issue by Vegemash** [microsoft/pyright#2085]
Maybe I'm doing this all wrong but it works in mypy. When a functions takes a `Type[TypeVar]`, the typevar seems to never get resolved.

**To Reproduce**
run pyright on the following code

**Expected behavior**
No errors and correctly revealed types

**Screenshots or Code**
If applicable, add screenshots or the text of the code (surrounded by triple back ticks) to help explain your problem.
```python
from typing import Type
from typing import TypeVar
from typing import Any


class A():
    def __init__(self, x: Any) -> None:
        pass

    def f(self) -> None:
        pass


T = TypeVar("T", str, int, A)


def factory(desired_type: Type[T]) -> T:
    return desired_type(1)


factory(str)
reveal_type(factory(str))  # mypy reports str, pyright reports T@factory

factory(int)
reveal_type(factory(int))  # mypy reports int, pyright reports T@factory

factory(A).f()
reveal_type(factory(A))  # mypy reports A, pyright reports T@factory
```
```
/tmp/parabol.py
  /tmp/parabol.py:21:9 - error: Argument of type "Type[str]" cannot be assigned to parameter "desired_type" of type "Type[T@factory]" in function "factory"
    Type "Type[str]" is incompatible with constrained type variable "T" (reportGeneralTypeIssues)
  /tmp/parabol.py:22:9 - error: Argument of type "Type[int]" cannot be assigned to parameter "desired_type" of type "Type[T@factory]" in function "factory"
    Type "Type[int]" is incompatible with constrained type variable "T" (reportGeneralTypeIssues)
  /tmp/parabol.py:23:9 - error: Argument of type "Type[A]" cannot be assigned to parameter "desired_type" of type "Type[T@factory]" in function "factory"
    Type "Type[A]" is incompatible with constrained type variable "T" (reportGeneralTypeIssues)
  /tmp/parabol.py:23:12 - error: Cannot access member "f" for type "str*"
    Member "f" is unknown (reportGeneralTypeIssues)
```
**VS Code extension or command-line**
command line @ 1.1.156

**Additional context**
If there is some way around this I'd love to know
<3 pyright, gotta go fast

**Comment by erictraut** [comment]
Thanks for the bug report. The constraint solver wasn't properly handling the case of `Type[T]` when `T` is a constrained type variable. This will be fixed in the next release.

**Comment by erictraut** [comment]
This fix is included in pyright 1.1.57, which I just published. It will also be included in the next release of pylance.

**Comment by Vegemash** [comment]
@erictraut Thank you! This was such a quick turn-around
